### PR TITLE
feat: Implement delete news article service layer - EXO-70383 - Meedsio/MIPs#119

### PR DESCRIPTION
**Pull request by sofyenne**

### CODE REVIEW

File: content-service/src/main/java/io/meeds/news/rest/NewsRestResourcesV1.java [hunk 1]
```diff
@@ -405,7 +405,7 @@ public Response getNewsById(@Context
       }
       org.exoplatform.services.security.Identity currentIdentity = ConversationState.getCurrent().getIdentity();
       News news = newsService.getNewsById(id, currentIdentity, editMode, newsObjectType);
-      if (news == null) {
+      if (news == null || (newsObjectType != null && NewsUtils.isDeletedNews(news.getId(), newsObjectType, currentIdentity.getUserId()))) {
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/rest/NewsRestResourcesV1.java]: we can instead add isDeleted attribute for News model

File: content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java [hunk 2]
```diff
@@ -338,12 +338,12 @@ public void deleteNews(String newsId, Identity currentIdentity, boolean isDraft)
                             false,
                             isDraft ? NewsObjectType.DRAFT.name().toLowerCase() : NewsObjectType.ARTICLE.name().toLowerCase());
     if (!news.isCanDelete()) {
-      throw new IllegalArgumentException("User " + currentIdentity.getUserId() + " is not authorized to delete news");
+      throw new IllegalAccessException("User " + currentIdentity.getUserId() + " is not authorized to delete news");
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java:338]: you can change also that for createNews and editNews cases, corresponding rest services to be tested for non authorized users

**Review by azayati — CHANGES_REQUESTED** [review]

File: content-service/src/main/java/io/meeds/news/rest/NewsRestResourcesV1.java (hunk 1 above)
Comment (sofyenne) [reply content-service/src/main/java/io/meeds/news/rest/NewsRestResourcesV1.java]: done !

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**